js/packages/react-ui/src/components/Charts/HorizontalBarChart/utils/HorizontalBarChartUtils.ts

```diff
@@ -0,0 +1,152 @@
+import { getDataKeys } from "../../utils/dataUtils";
+import { getCanvasContext } from "../../utils/styleUtils";
+import { HorizontalBarChartVariant } from "../types";
+
+export const BAR_HEIGHT = 16;
+export const BAR_GAP = 10;
+
+/**
+ * Get the maximum text width for category labels
+ * @param data - The data to be displayed in the chart
+ * @param categoryKey - The key of the category to be displayed in the chart
+ * @returns The maximum width needed for category labels
+ */
+export const getMaxCategoryLabelWidth = (
+  data: Array<Record<string, string | number>>,
+  categoryKey: string,
+): number => {
+  if (data.length === 0) {
+    return 100; // Default fallback
+  }
+
+  const context = getCanvasContext();
+  if (!context) {
+    // Fallback for SSR or when canvas is not available
```

Comment:
```suggestion
    // Fallback when canvas is not available
```
we don't support ssr

Reply:
as a matter of fact, canvas will always be available